apigee/edge_client.go

```diff
@@ -237,7 +237,7 @@ func NewEdgeClient(o *EdgeClientOptions) (*EdgeClient, error) {
 
 	if !o.Auth.SkipAuth {
 		var e error
-		if o.Auth == nil || (o.Auth.Password == "" && o.Auth.BearerToken == "") {
+		if o.Auth.Password == "" && o.Auth.BearerToken == "" {
```

Comment:
With removing the `o.Auth == nil `check - is there any chance that o.Auth itself could be nil?

Reply:
My thought was o.Auth would be validated by `shared.Resolve()`. I can add a nil check here. But it need to be pointed out that it will needed be added along with `!o.Auth.SkipAuth` - the current one doesn't prevent a runtime panic at all.

Reply:
That's fine. Just wanted to make sure we didn't open a hole. :)